src/test/java/teammates/test/cases/action/AdminActivityLogPageActionTest.java

```diff
@@ -1,27 +1,551 @@
+     *
+     * <p>expectedLogs is a 2D array, the outer indices correspond to {@link #LOG_MESSAGE_INDEX_TODAY}
+     * {@link #LOG_MESSAGE_YESTDAY_INDEX} and {@link #LOG_MESSAGE_INDEX_TWO_DAYS_AGO}, the inner indices for
+     * every {@code LOG_MESSAGE_*_INDEX} correspond to the orders in the test data.
+     *
+     * @param expectedLogs
+     * @param actualLogs
+     */
+    private void verifyLogs(int[][] expectedLogs, List<ActivityLogEntry> actualLogs) {
+        List<String> expectedMsgs = generateExpectedMsgFrom(expectedLogs);
+
+        assertEquals(expectedMsgs.size(), actualLogs.size());
+        for (int i = 0; i < expectedMsgs.size(); i++) {
+            String actualMsg = actualLogs.get(i).generateLogMessage();
+            actualMsg = actualMsg.replace("<mark>", "").replace("</mark>", "");
+            assertTrue("expecte: " + expectedMsgs.get(i) + "to contain:" + actualMsg,
+                       expectedMsgs.get(i).contains(actualMsg));
+        }
+    }
+
+    private List<String> generateExpectedMsgFrom(int[][] expectedLogs) {
+        List<String> result = new ArrayList<String>();
+        for (int i = 0; i < expectedLogs.length; i++) {
+            for (int j = 0; j < expectedLogs[i].length; j++) {
+                result.add(logMessages.get(i).get(expectedLogs[i][j]));
+            }
+        }
+        return result;
+    }
+
+    private void verifyStatusMessage(String message, int totalLogs, int filteredLogs, Date earliestDateInUtc) {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy HH");
+        sdf.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));
+
+        assertTrue(message.contains("Total Logs gone through in last search: " + totalLogs));
+        assertTrue(message.contains("Total Relevant Logs found in last search: " + filteredLogs));
+        assertTrue(message.contains("Logs are from following version(s): 1"));
+        assertTrue(message.contains("All available version(s): 1"));
+        // bug might be introduced when the time is 00:00 AM.
+        assertTrue(message.contains("The earliest log entry checked on <b>" + sdf.format(earliestDateInUtc.getTime())));
+    }
+
+    private void verifyLocalTimeInStatusMessage(String message, Date timeInUtc, double localTimeZone) {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy HH");
+        Calendar timeLocal = TimeHelper.convertToUserTimeZone(TimeHelper.dateToCalendar(timeInUtc), localTimeZone);
+        assertTrue(message.contains(sdf.format(timeLocal.getTime())));
+        assertTrue(message.contains(String.format("in Local Time Zone (%.1f)", localTimeZone)));
+    }
+
+    private void verifyLoadingLocalTimeAjaxResult(String expected, String role, String googleId, long timeInMillis) {
+        String[] params = new String[]{"logRole", role, "logGoogleId", googleId,
+                                       "logTimeInAdminTimeZone", String.valueOf(timeInMillis)};
```

Comment:
Use 8-space indentation for line wrapping. We are moving away from custom indentation levels. But you can choose the line break location that gives the most readable code.